features/step_definitions/authentication.js

```diff
@@ -17,7 +17,7 @@ var apiSteps = function () {
           return false;
         }
 
-        if (this.currentUser === null) {
+        if (this.currentUser === null || this.currentUser === undefined) {
```

Comment:
if user was logged-in, then logged-out, this.currentUser has been unset is. set to "undefined"